src/sage/matroids/chow_ring.py

```diff
@@ -0,0 +1,389 @@
+
+        EXAMPLES::
+
+            sage: M1 = matroids.Uniform(2,5)
+            sage: ch = M1.chow_ring(QQ, False)
+            sage: ch._latex_()
+            '\\Bold{Q}[A_{0}, A_{1}, A_{2}, A_{3}, A_{4}, A_{01234}] / I_{M} + J_{M} of matroid \\begin{array}{l}\n\\text{\\texttt{U(2,{ }5):{ }Matroid{ }of{ }rank{ }2{ }on{ }5{ }elements{ }with{ }circuit{-}closures}}\\\\\n\\text{\\texttt{{\\char`\\{}2:{ }{\\char`\\{}{\\char`\\{}0,{ }1,{ }2,{ }3,{ }4{\\char`\\}}{\\char`\\}}{\\char`\\}}}}\n\\end{array}'
+        """
+        from sage.misc.latex import latex
+        return "{} / {}".format(latex(self._ideal.ring()), latex(self._ideal))
+
+    def _coerce_map_from_base_ring(self):
+        r"""
+        Disable the coercion from the base ring from the category.
+
+        TESTS::
+
+            sage: ch = matroids.Wheel(3).chow_ring(QQ, False)
+            sage: ch._coerce_map_from_base_ring() is None
+            True
+        """
+        return None  # don't need anything special
+
+    def basis(self):
+        r"""
+        Return the monomial basis of the given Chow ring.
+
+        EXAMPLES::
+
+            sage: ch = matroids.Uniform(3, 6).chow_ring(QQ, True, 'fy')
+            sage: ch.basis()
+            Family (1, B1, B1*B012345, B0, B0*B012345, B01, B01^2, B2,
+            B2*B012345, B02, B02^2, B12, B12^2, B3, B3*B012345, B03, B03^2,
+            B13, B13^2, B23, B23^2, B4, B4*B012345, B24, B24^2, B34, B34^2,
+            B04, B04^2, B14, B14^2, B5, B5*B012345, B25, B25^2, B35, B35^2,
+            B45, B45^2, B05, B05^2, B15, B15^2, B012345, B012345^2,
+            B012345^3)
+            sage: set(ch.defining_ideal().normal_basis()) == set(ch.basis())
+            True
+            sage: ch = matroids.catalog.Fano().chow_ring(QQ, False)
+            sage: ch.basis()
+            Family (1, Abcd, Aace, Aabf, Adef, Aadg, Abeg, Acfg, Aabcdefg,
+            Aabcdefg^2)
+            sage: set(ch.defining_ideal().normal_basis()) == set(ch.basis())
+            True
+            sage: ch = matroids.Wheel(3).chow_ring(QQ, True, 'atom-free')
+            sage: ch.basis()
+            Family (1, A0, A0*A012345, A2, A2*A012345, A3, A3*A012345, A23,
+            A23^2, A1, A1*A012345, A013, A013^2, A4, A4*A012345, A04, A04^2,
+            A124, A124^2, A5, A5*A012345, A025, A025^2, A15, A15^2, A345,
+            A345^2, A012345, A012345^2, A012345^3)
+            sage: set(ch.defining_ideal().normal_basis()) == set(ch.basis())
+            True
+        """
+        flats = [X for i in range(1, self._matroid.rank() + 1)
+                 for X in self._matroid.flats(i)] #Non empty flats
+        flats_gen = self._ideal.flats_generator()
+        R = self._ideal.ring()
+        flats = sorted(flats, key=lambda X: (len(X), sorted(X)))
+        ranks = {F: self._matroid.rank(F) for F in flats}
+        monomial_basis = []
+        reln = lambda x,y: x <= y
+        lattice_flats = Poset((flats, reln))
```

Comment:
```suggestion
        F = self._matroid.lattice_of_flats()
        H = F.hasse_diagram()
        H.delete_vertex(M.flats(0)[0])  # remove the empty flat
        lattice_flats = Poset(H)
        flats = list(lattice_flats)
        flats_gen = self._ideal.flats_generator()
        R = self._ideal.ring()
        flats.sort(key=lambda X: (len(X), sorted(X)))
        ranks = {F: self._matroid.rank(F) for F in flats}
        monomial_basis = []
```
I think this will be faster. It is also (uses the) caches the full lattice of flats in the matroid.